File: backend/api-productos/src/utils/authMiddleware.js
const jwt = require('jsonwebtoken');

module.exports = {
  validateToken(authHeader) {
    if (!authHeader) {
      throw new Error('Authorization token is required.');
    }

    const token = authHeader.replace('Bearer ', '');
    const decoded = jwt.verify(token, process.env.JWT_SECRET);

    if (!decoded.tenantId) {
      throw new Error('Invalid token payload: missing tenantId');
    }

    return decoded;
  },
};

File: backend/api-compras/src/utils/authMiddleware.py
import jwt
import os

def validate_token(auth_header):
    if not auth_header:
        raise Exception("Authorization token is required.")
    token = auth_header.replace("Bearer ", "")
    secret = os.environ.get("JWT_SECRET")
    if not secret:
        raise Exception("JWT_SECRET environment variable not set.")
    return jwt.decode(token, secret, algorithms=["HS256"]) 